zebra/zebra_vty.c

```diff
@@ -419,6 +419,33 @@ static void show_nexthop_detail_helper(struct vty *vty,
 	}
 }
 
+static void zebra_show_ip_route_opaque(struct vty *vty, struct route_entry *re,
+				       struct json_object *json)
+{
+	if (!re->opaque)
+		return;
+
+	switch (re->type) {
+	case ZEBRA_ROUTE_SHARP:
+		if (json)
+			json_object_string_add(json, "opaque",
+					       (char *)re->opaque->data);
+		else
+			vty_out(vty, "    Opaque Data: %s",
+				(char *)re->opaque->data);
+		break;
+	case ZEBRA_ROUTE_BGP:
+		if (json)
+			json_object_string_add(json, "asPath",
+					       (char *)re->opaque->data);
+		else
+			vty_out(vty, "    AS-Path: %s",
+				(char *)re->opaque->data);
```

Comment:
If we are planning to add more opaque data not only as-path, how we gonna distinguish it from `data`?

Reply:
when that work is done we'll add a structure.  At this point it is not needed